.github/workflows/changelog-check.yaml

```diff
@@ -0,0 +1,31 @@
+name: Changelog Checker
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+    branches:
+      - main
+
+
+jobs:
+  changelog-check:
+    if: "!contains(github.event.pull_request.labels.*.name, 'pr/no-changelog')"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't check out all branches
+      - name: Check for changelog entry in diff
+        run: |
+          # Check if there's a diff on CHANGELOG.md
+          changelog_diff=$(git --no-pager diff --name-only HEAD "$(git merge-base HEAD "origin/${{ github.event.pull_request.base.ref }}")" -- CHANGELOG.md)
```

Comment:
If/when we come to roll this out to lots of repos, we might want to package this logic up into an internal reusable action to reduce maintenance burden.

Reply:
Definitely!